src/http_microhttpd.c

```diff
@@ -757,9 +844,11 @@ static int show_splashpage(struct MHD_Connection *connection, t_client *client)
 	char *imagesdir = NULL;
 	char *pagesdir = NULL;
 
+	tmpl_init_templor(&templor);
+
 	memset(redirect_url_encoded, 0, sizeof(redirect_url_encoded));
 	redirect_url = get_redirect_url(connection);
-	if (redirect_url) {
+	if (strlen(redirect_url) > 0) {
```

Comment:
this is not a good idea, because get_redirect_url can return a NULL pointer. you can check both or just put the strlen() check into get_redirect_url and return NULL when strlen() == 0.